jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -504,7 +504,11 @@ YUI.add('juju-env-api', function(Y) {
         }
         // Clean up for log out text.
         this.failedAuthentication = false;
-      } else {
+      } else if (
+        !data.error || data.error.indexOf('redirection required') === -1) {
```

Comment:
There you go, second instance of the same error, an external function will be already useful!